### ISSUE TITLE
Accessibility improvements for modal dialog

### ISSUE THREAD

**Issue by TomFranssen** [DivanteLtd/vue-storefront#3907]
## What is the motivation for adding / enhancing this feature?

When opening a modal on VSF for instance the Log In modal and you tab through the modal till the end and press tab one more time your browser automatically starts it's focus on the top of the website instead of the top of the modal. This way it's very hard to get the focus back in the modal again. 
According to the official WCAG standard it's better to start again at the top of the modal when pressing tab again at the end of the modal. You can see an example here: https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html Bootstrap 4 also does this very well by default.

## What are the acceptance criteria 

- A modal dialog should always contain the focus of the user
- The user should not be able to focus the elements on the background.

## Can you complete this feature request by yourself?

- [ ] YES
- [X] NO

## Which [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html) state this refers to? Info for developer.
Pick one option.

- [x] This is a normal feature request. This should be available on https://test.storefrontcloud.io and then after tests this can be added to next Vue Storefront version. In this case Developer should create branch from `develop` branch and create Pull Request `2. Feature / Improvement` back to `develop`.
- [ ] (**Pick this option only if you're sure**) This is an important improvement request for current Release Candidate version on https://next.storefrontcloud.io and should be placed in next RC version. In this case Developer should create branch from `release` branch and create Pull Request `3. Stabilisation fix` back to `release`.
- [ ] (**Pick this option only if you're sure**) This is a critical improvement request for current Stable version on https://demo.storefrontcloud.io and should be placed in next stable version. In this case Developer should create branch from `hotfix` or `master` branch and create Pull Request `4. Hotfix` back to `hotfix`.
